### ISSUE TITLE
Organize and document dependencies

### ISSUE THREAD

**Issue by drewoldag** [lincc-frameworks/python-project-template#109]
**tl;dr;** 
There are dependencies defined in multiple places, how do we keep these in sync.

**The details**
Currently there are three areas where dependencies live in the template:

1. pyproject.toml `dependencies`
2. pyproject.toml `project.optional-dependencies`
3. ./docs/requirements.txt

The list under `dependencies` are the packages that will be required for an end user of a project to install and run the project.
The list under project.optional-dependencies` are the packages that are required for developers to work on the project
The list in ./docs/requirements.txt are the packages necessary for sphinx and readthedocs to build the documentation. 

If a developer wants to build the docs locally, they need to install the packages in ./docs/requirements.txt. This is perhaps, not intuitive. Thus when I first introduced sphinx, I duplicated the packages in ./docs/requirements.txt in `project.optional-dependencies:dev`. 

Now that we have jupyter notebooks that can be build automatically, there are a handful of additional packages required in order to build the documentation locally, and it is becoming clear that neither we nor the user of the hydrated template would want to remember to duplicate the packages in two locations.

Unfortunately requirements.txt doesn't seem to support references to pyproject.toml, nor does pyproject.toml's `project.optional-dependencies` section support references to files.

**Comment by drewoldag** [comment]
After some discussion with @OliviaLynn I believe that we're in agreement that simple is better here, and that we should allow users to manually duplicate dependencies in `.../docs/requirements.txt` and `pyproject.toml:project.optional-dependencies`.

For the time being adding some comments in both places specifying that "if you add something here, you'll need to add it there" will probably suffice until/if we can find a more elegant solution. 

Also for broader visibility, I had considered breaking the optional-dependencies section of pyproject.toml into a [dev] and [doc] section. However I no longer think that's a good idea because:

1. Any person developing the code will (theoretically) build the sphinx documentation as part of the pre-commit hooks, and so will need to all of the [doc] dependencies installed
2. Any person editing just the documentation will still commit those changes via a git commit, which will trigger a pre-commit and which will require the dependencies in [dev]
3. Separating the two adds a new branch to the possible template configurations that need to be tested and maintained.

**Comment by drewoldag** [comment]
So for the time being, I believe that the action items here are:

1. Add comments to requirements.txt and pyproject.toml about keeping the dependencies in sync
2. Make sure that all the dependencies that are currently defined in docs/requirements are also in optional-dependencies.
